SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java

```diff
@@ -297,13 +298,21 @@ public SingularityPendingRequestParent scheduleImmediately(@ApiParam("The reques
     Optional<Boolean> skipHealthchecks = Optional.absent();
     Optional<List<String>> commandLineArgs = Optional.absent();
     Optional<Resources> resources = Optional.absent();
+    Optional<Long> runAt = Optional.absent();
 
     if (maybeRunNowRequest.isPresent()) {
       message = maybeRunNowRequest.get().getMessage();
       runId = maybeRunNowRequest.get().getRunId();
       skipHealthchecks = maybeRunNowRequest.get().getSkipHealthchecks();
       commandLineArgs = maybeRunNowRequest.get().getCommandLineArgs();
       resources = maybeRunNowRequest.get().getResources();
+      runAt = maybeRunNowRequest.get().getRunAt();
+
+      if (runAt.isPresent() && runAt.get() > (System.currentTimeMillis() + TimeUnit.DAYS.toMillis(30))) {
```

Comment:
Should have been more specific in my comment. With any hard limits, we always try to make them configurable via parameters somewhere in `SingularityConfiguration`, that way if for some reason we need to override, we can stick it in the yaml and be gtg